### PR TITLE
Feat: fetch Github Installation Id from env0

### LINKED ISSUE #928: Fetch Github Installation Id from env0
**Is your feature request related to a problem? Please describe.**
Right now customers who wish to use native github vcs templates, instead of "other vcs" need to provide the github-installation-id and this is always a pain to get or save it somewhere.

**Describe the solution you'd like**
We'd like to make this easier for customers, so I added a function, to fetch the github installation id (in the future this can be expanded for other VCSs tokens)
The function gets 3 inputs: 
 - `organizationId`
 - `repository` - the github repository where the env0 application is installed on
 - `vcsType` - currently only `github` is supported

The new function is accessible so:
method: `GET`
URL: `vcs-token/{vcsType}`
Request Parameters: `organizationId`, `repository`
Path Parameter: `vcsType` - currently only github is supported

Please notice the repository has to be url-encoded.

here's a CURL example of a request:
```
curl --location 'https://api-pr16600.dev.env0.com/vcs-token/github?repository=https%3A%2F%2Fgithub.com%2Fsagydr%2Fenv0_templates&organizationId=bdf09db8-25e1-4e0d-bd80-35ca7fe70243' \
--header 'Authorization: Basic <token>'
```

The [RND ticket](https://trello.com/c/42YzPCQM) shows an example how using that new function can be easier to set a github template.

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
resolves #928 

### Solution

1. Added the API call.
2. Added unit tests for API call.
3. Added the data source.
4. Added acceptance tests for the data source.
5. Added harness test for the data source.
6. Added an example of the data source.

### CODE REVIEW

File: client/api_client.go [hunk 1]
```diff
@@ -166,6 +166,7 @@ type ApiClientInterface interface {
 	CloudAccountDelete(id string) error
 	CloudAccount(id string) (*CloudAccount, error)
 	CloudAccounts() ([]CloudAccount, error)
+	VscToken(vcsType string, repository string) (*VscToken, error)
```
Comment (sagydr) [inline client/api_client.go:169]: Vcs not Vsc

File: env0/data_github_installation_id.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+			"repository": {
+				Type:        schema.TypeString,
+				Description: "the name of the repository",
+				Required:    true,
+			},
+			"github_installation_id": {
+				Type:        schema.TypeInt,
+				Description: "the github installation id",
+				Computed:    true,
+			},
```
Comment (sagydr) [inline env0/data_github_installation_id.go:27]: why both? "repository" is parameter and "github installation id" is the return type?

**Review by sagydr — APPROVED** [review]
looks fine, approved with small questions

File: env0/data_github_installation_id.go (hunk 2 above)
Comment (TomerHeber) [reply env0/data_github_installation_id.go:27]: correct. One is an input, the other is computed.

File: client/api_client.go (hunk 1 above)
Comment (TomerHeber) [reply client/api_client.go:169]: I'll fix that. Thanks!

**Pull request merged**